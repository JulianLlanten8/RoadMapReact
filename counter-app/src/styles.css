html,
body {
  background-color: #242424;
  color: white;
  font-family: Helvetica, sans-serif;
  font-size: 1.3 rem;
  padding: 70px;
}

button {
  background-color: #242424;
  border: 1px solid white;
  color: white;
  font-size: 1.3 rem;
  padding: 10px;
  margin: 10px;
  cursor: pointer;
}
