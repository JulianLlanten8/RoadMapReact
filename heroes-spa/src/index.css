.zoom {
  transition: transform 0.5s ease-in-out;
}

.zoom:hover {
  transform: scale(1.1);
  z-index: 2;
}

@keyframes show {
  from {
    opacity: 0;
    scale: 0.5;
  }
  to {
    opacity: 1;
    scale: 100%;
  }
}
body {
  background-color: oklch(26.03% 0 0);
}
.card-blur {
  backdrop-filter: blur(12px) saturate(180%);
  -webkit-backdrop-filter: blur(12px) saturate(180%);
  background-color: rgba(3, 118, 242, 0.6);
  border-radius: 12px;
  border: 1px solid rgba(255, 255, 255, 0.125);

  /* animation entry */
  view-timeline-name: --image;
  vie-timeline-axis: block;

  animation-timeline: --image;
  animation-name: show;
  animation-range: entry 5% cover 30%;
  animation-fill-mode: both;
}
