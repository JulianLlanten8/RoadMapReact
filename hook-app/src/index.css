body {
  padding: 20px;
}
button {
  margin-right: 10px;
}

a {
  color: #61dafb;
  margin-right: 5px;
}
