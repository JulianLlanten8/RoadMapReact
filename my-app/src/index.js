import { heroes } from "./data/heroes";

console.log(heroes);

const getHeroeById = (id) => {
  return heroes.find((heroe) => heroe.id === id);
};

console.log(getHeroeById(2));
