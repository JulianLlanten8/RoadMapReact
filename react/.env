REACT_APP_API_KEY_GIPHY = "n7X50L3fGzjK3N84gA7yPI6PLxzrIhS0"
