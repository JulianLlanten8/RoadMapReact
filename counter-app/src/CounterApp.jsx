import { useState } from "react";
import PropTypes from "prop-types";

export function CounterApp({ value }) {
  const [counter, setCounter] = useState(value);

  const handleAdd = () => setCounter(counter + 1);

  const handleDecrease = () =>
    counter > 0 ? setCounter(counter - 1) : setCounter(0);

  const handleReset = () => setCounter(value);

  return (
    <>
      <h1>CounterApp</h1>
      <h2>{counter} </h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleDecrease}>-1</button>
      <button onClick={handleReset} aria-label="btn-reset">Reset</button>
    </>
  );
}

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
