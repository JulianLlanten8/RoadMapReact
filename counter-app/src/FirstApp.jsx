import { Fragment } from "react";
import PropTypes from "prop-types";
/* 
  En react siempre se debe retornar un nodo padre y dentro de este los hijos 
  algo parecido a vue que en vue se debe retornar un solo elemento.
*/
class persona {
  constructor(nombre, apellido, edad) {
    this.nombre = nombre;
    this.apellido = apellido;
    this.edad = edad;
    this.nombreCompleto = `${nombre} ${apellido}`;
  }
  info() {
    return `${this.nombreCompleto} Edad: ${this.edad}`;
  }
}

const hugo = new persona("Hugo", "Julian", 26);

const FirstApp = () => (
  <Fragment>
    <p>{hugo.info()}</p>
    <p>Frontend Developer</p>
  </Fragment>
);
// Otra forma de hacerlo y la habitual
const SecondApp = ({ title = "No tiene" }) => {
  // console.log(props);
  return (
    <>
      <h2>Fragmento de forma diferente</h2>
      <p>{title}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
};

export { FirstApp, SecondApp };
