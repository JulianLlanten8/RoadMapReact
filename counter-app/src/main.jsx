import React from "react";
import ReactDOM from "react-dom/client";
// import { HelloWorld } from "./HelloWorld";
import { FirstApp, SecondApp } from "./FirstApp";
// import { CounterApp } from "./CounterApp";
import "./styles.css";

// Se renderiza el componente App en el elemento con id root
ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <HelloWorld /> */}
    <FirstApp />
    <SecondApp title="Hola, soy Vegueta" subTitle="Sub titulo" />
    {/* <CounterApp value={0} /> */}
  </React.StrictMode>
);
