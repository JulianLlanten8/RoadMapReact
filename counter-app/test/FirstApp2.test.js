import { render, screen } from "@testing-library/react";
import { SecondApp } from "../src/FirstApp";

describe("Puerabas en <SecondApp/>", () => {
  const title = "Hola soy Goku";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<SecondApp title={title} />);
    console.log(container.innerHTML);
    expect(container).toMatchSnapshot(); // Crea el snapshot y lo compara con el actual
  });

  test('Debe de mostrar el mensaje "Hola, soy goku', () => {
    render(<SecondApp title={title} />);
    expect(screen.getByText(title)).toBeTruthy();
  });

  test("Debe de mostrar el titulo en un h1", () => {
    render(<SecondApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toBe(title);
  });
  test("Debe mostrar el subtitulo enviado por props", () => {
    render(<SecondApp title={title} subTitle="Soy un subtitulo" />);
    expect(screen.getAllByText("Soy un subtitulo").length).toBe(2); //en el componente se repite el subtitulo
  });
});
