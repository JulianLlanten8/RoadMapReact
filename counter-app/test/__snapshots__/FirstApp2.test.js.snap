// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puerabas en <SecondApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h2>
    Fragmento de forma diferente
  </h2>
  <h1
    data-testid="test-title"
  >
    Hola soy Goku
  </h1>
  <p>
    El subtitulo por defecto
  </p>
  <p>
    El subtitulo por defecto
  </p>
</div>
`;
